### PR TITLE
editoast: sync schema with api

### PR DESCRIPTION
**Pull request by flomonster**

### CODE REVIEW

File: api/osrd_infra/schemas/infra.py [hunk 1]
```diff
@@ -477,29 +492,29 @@ def register_extension(extension):
 @register_extension(object=TrackSection, name="sncf")
 class TrackSectionSncfExtension(BaseModel):
     line_code: int = Field(description="Code of the line used by the corresponding track section")
-    line_name: constr(max_length=255) = Field(description="Name of the line used by the corresponding track section")
```
Comment (Khoyo) [inline api/osrd_infra/schemas/infra.py:480]: You are removing the max length constraint here and in a few other places too. Is that voluntary?

File: api/osrd_infra/schemas/infra.py [hunk 2]
```diff
@@ -12,7 +20,8 @@
 # Traits
 # Used as an input model in the definition of the following classes.
 
-Identifier = NewType("Identifier", constr(max_length=255))
+Identifier = NewType("Identifier", constr(min_length=1, max_length=255))
+NotBlankStr = NewType("NotBlankStr", constr(min_length=1))
```
Comment (Khoyo) [inline api/osrd_infra/schemas/infra.py]: NonBlankStr? NonblankStr?

File: editoast/src/schema/deserializer.rs [hunk 3]
```diff
@@ -0,0 +1,146 @@
+use std::ops::{Deref, DerefMut};
+
+use rand::{distributions::Alphanumeric, thread_rng, Rng};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// A wrapper around a String that ensures that the string is not empty and not longer than 255 characters.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Identifier(String);
```
Comment (Khoyo) [inline editoast/src/schema/deserializer.rs]: Yay for semantic typing \o/ 🎉

**Review by multun — DISMISSED** [review]

File: api/osrd_infra/schemas/infra.py (hunk 1 above)
Comment (flomonster) [reply api/osrd_infra/schemas/infra.py:480]: Yes, we don't have a reason to constraint these fields.

**Pull request merged**